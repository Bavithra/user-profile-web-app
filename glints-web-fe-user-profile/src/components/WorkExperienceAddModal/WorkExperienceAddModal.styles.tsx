import styled from 'styled-components';
import { FontWeights } from '../../styles/FontWeights';

export const WorkExperienceAddModalContainer = styled.div`
  display: flex;
  padding: 30px 10px;
  height: 100%;
  flex-direction: column;
  overflow: hidden;
`;

export const WorkExperienceAddModalInfo = styled.div`
  display: flex;
  padding: 20px;
  height: 100%;
  width: 100%;
  font-size: 14px;
  flex-direction: column;
`;

export const Header = styled.div`
  font-size: 16px;
  font-weight: ${FontWeights.bold};
  line-height: 43px;
  color: rgb(55, 22, 80);
`;

export const InputContainer = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
`;
