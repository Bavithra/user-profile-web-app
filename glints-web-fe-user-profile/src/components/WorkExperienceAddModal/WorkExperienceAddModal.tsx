import React, { SetStateAction, useCallback } from "react";
import Button from "../../common/Button";
import DatePicker from "../../common/DatePicker";

import Modal from "../../common/Modal";
import TextInput from "../../common/TextInput";
import { MAX_TEXTAREA_INPUT } from "../../Constants";
import { Text } from "../../styles/Common.styles";
import { WorkExperience } from "../../types/WorkExperience.type";
import DateTimeUtil from "../../utils/DateTimeUtil";
import WorkExperienceUtil from "../../utils/WorkExperienceUtil";

import {
  WorkExperienceAddModalContainer,
  WorkExperienceAddModalInfo,
  Header,
  WorkExperienceAddModalFooter,
  PresentCompany,
  Company,
  CompanyLogoTitle,
  TextArea,
} from "./WorkExperienceAddModal.styles";

type Props = {
  isOpen: boolean;
  workExperience: WorkExperience;
  setWorkExperience: React.Dispatch<SetStateAction<WorkExperience>>;
  setIsWorkExperienceModalOpen: (shouldShowModal: boolean) => void;
  onSaveButtonClick: (workExperience: WorkExperience) => void;
};

export default function WorkExperienceAddModal(props: Props) {
  const {
    isOpen,
    workExperience,
    setIsWorkExperienceModalOpen,
    setWorkExperience,
    onSaveButtonClick,
  } = props;

  function onCancelClick() {
    setIsWorkExperienceModalOpen(false);
  }

  const updateWorkExperience = useCallback(
    (inputKey: keyof WorkExperience, value: string) => {
      setWorkExperience((previousInputs) => (WorkExperienceUtil.update(previousInputs, inputKey, value)));
    },
    [setWorkExperience]
  );

  function onActionButtonClick(workExperience?: WorkExperience) {
    if (workExperience !== undefined) {
      onSaveButtonClick(workExperience);
    }

    setWorkExperience(WorkExperienceUtil.getInitialWorkExperienceInput());
    onCancelClick();
  }

  function setCompanyDetails(value: string): void {
    updateWorkExperience("company", value);
    updateWorkExperience(
      "company-logo",
      WorkExperienceUtil.getCompanyLogoUrl(value)
    );
  }

  return (
    <Modal isOpen={isOpen} onClose={onCancelClick}>
      <WorkExperienceAddModalContainer>
        <WorkExperienceAddModalInfo>
          <Header>ADD WORK EXPERIENCE</Header>

          <DatePicker
            selectedDate={DateTimeUtil.getDate(workExperience["start-date"])}
            placeholder={"Start Date*"}
            isMonthPicker={true}
            onChange={(value) =>
              updateWorkExperience(
                "start-date",
                DateTimeUtil.getDisplayDateWithDay(value!)
              )
            }
          />

          <DatePicker
            selectedDate={DateTimeUtil.getDate(workExperience["end-date"])}
            placeholder={"End Date*"}
            minDate={new Date(workExperience["start-date"])}
            isMonthPicker={true}
            onChange={(value) =>
              updateWorkExperience(
                "end-date",
                DateTimeUtil.getDisplayDateWithDay(value!)
              )
            }
          />
          <PresentCompany>
            <input
              type="checkbox"
              checked={
                DateTimeUtil.getDisplayDateWithDay(new Date()) ===
                workExperience["end-date"]
              }
              onChange={(event) =>
                updateWorkExperience(
                  "end-date",
                  DateTimeUtil.getEndDate(
                    event.target.checked,
                    workExperience["end-date"]
                  )
                )
              }
            />
            <Text>I'm currently working in this company</Text>
          </PresentCompany>

          <TextInput
            label="Job title*"
            value={workExperience["job-title"]}
            onChange={(value) => updateWorkExperience("job-title", value)}
          />

          <Company>
            <TextInput
              label="Company*"
              value={workExperience.company}
              onChange={(value) => setCompanyDetails(value)}
            />
            <CompanyLogoTitle>
              logo
              <img
                alt="logo"
                src={workExperience["company-logo"]}
                onError={(e) => {
                  updateWorkExperience(
                    "company-logo",
                    "https://plugins.jetbrains.com/files/19441/190795/icon/pluginIcon.svg"
                  );
                }}
              />
            </CompanyLogoTitle>
          </Company>

          <TextArea
            placeholder={"Job description(optional)"}
            maxLength={MAX_TEXTAREA_INPUT}
            value={workExperience["job-description"]}
            onChange={(value) =>
              updateWorkExperience("job-description", value.target.value)
            }
          />
        </WorkExperienceAddModalInfo>

        <WorkExperienceAddModalFooter>
          <Button onClick={() => onActionButtonClick()}>CANCEL</Button>
          <Button
            isDisable={WorkExperienceUtil.shouldSaveButtonDisable(
              workExperience
            )}
            onClick={() => onActionButtonClick(workExperience)}
          >
            SAVE
          </Button>
        </WorkExperienceAddModalFooter>
      </WorkExperienceAddModalContainer>
    </Modal>
  );
}

