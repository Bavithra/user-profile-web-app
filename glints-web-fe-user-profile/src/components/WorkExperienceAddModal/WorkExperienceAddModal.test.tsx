import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import WorkExperienceAddModal from "./WorkExperienceAddModal";

describe('WorkExperienceAddModal Block', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  it('should render correctly', () => {
    const onSaveButtonClick = jest.fn();
    const setIsWorkExperienceModalOpen = jest.fn();
    const setWorkExperience = jest.fn();

    const component = renderer.create(<WorkExperienceAddModal
      isOpen
      workExperience={{
        id: "",
        'start-date': 'July 2022',
        'end-date': 'July 2022',
        'job-title': 'Engineer',
        company: 'Glints',
        'company-logo': '',
        'job-description': 'Glints Desc',
      }}
      setIsWorkExperienceModalOpen={setIsWorkExperienceModalOpen}
      setWorkExperience={setWorkExperience}
      onSaveButtonClick = {onSaveButtonClick}
      />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});


