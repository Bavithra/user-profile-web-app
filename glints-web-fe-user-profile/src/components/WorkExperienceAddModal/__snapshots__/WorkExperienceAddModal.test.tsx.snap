// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperienceAddModal Block should render correctly 1`] = `null`;
